packages/ui/react-ui-editor/src/extensions/comments.ts

```diff
@@ -592,27 +603,41 @@ const hasActiveSelection = (state: EditorState): boolean => {
   return state.selection.ranges.some((range) => !range.empty);
 };
 
-/**
- * Update comments state field.
- */
-export const useComments = (view: EditorView | null | undefined, id: string, comments?: Comment[]) => {
-  useEffect(() => {
-    if (view) {
-      // Check same document.
-      // NOTE: Hook might be called before editor state is updated.
+class ExternalCommentSync implements PluginValue {
+  private unsubscribe: () => void;
+
+  // TODO(Zan): How to achieve this with deep signal?
+  constructor(view: EditorView, id: string, threads: ReadonlySignal<ThreadType[]>) {
+    this.unsubscribe = effect(() => {
```

Comment:
I think it would be good to try to do this without directly depending on signals in the extension. Could it take a subscribe/get pair of callbacks?

Reply:
Yeah that felt weird to me! Good call.